### DISCUSSION TITLE
Is it possible to use Cypress for a Tauri Desktop app?

### DISCUSSION THREAD

**Discussion by Nickog13** [cypress-io/cypress#24838] [Questions and Help]
Hi All,

As I am currently trying to apply test automation on a Tauri desktop app, I am wondering if this is possible using cypress.
In my opinion it is probably not possible without major and undesired changes in the app itself to use the browser as "proxy" between the front-end and the Tauri backend.

Has anyone tried to use cypress for a Tauri desktop app yet or has anyone gained experience in a similar issue, e.g with an Electron Desktop app?

**Comment by glyad** [comment]
Hi @Nickog13,

I'm using Cypress to test my frontend part of the Tauri-based app. Yes, unfortunately, I can't use Cypress to test the whole app end-to-end, because we lack support on the side of Cypress. Anyway, it works well for me, since I need the so-called backend as a pipe to the rest of the world only, and all of the business logic is implemented as a part of the front-end's Model layer. My MVVM architecture has an additional layer called the Data Providers layer. I have at least two implementations of the particular provider interface: fake and real. I use the DI container to register the appropriate provider implementation depending on the availability of the Tauri IPC. So, in the browser (at the test) the app uses the fake data.
